index.js

```diff
@@ -0,0 +1,91 @@
+let slideIndex = 0;
+
+function moveSlide(n) {
+  const slides = document.querySelectorAll('.carousel__item');
+  const totalSlides = slides.length;
+
+  slideIndex += n;
+
+  if (slideIndex >= totalSlides) {
+    slideIndex = 0;
+  } else if (slideIndex < 0) {
+    slideIndex = totalSlides - 1;
+  }
+
+  updateSlidePosition();
+  updateDots();
+}
+
+function currentSlide(n) {
+  slideIndex = n;
+  updateSlidePosition();
+  updateDots();
+}
+
+function updateSlidePosition() {
+  const slides = document.querySelectorAll('.carousel__item');
+  const slideWidth = slides[0].offsetWidth;
+  const offset = -slideIndex * slideWidth;
+  document.querySelector('.carousel__img').style.transform = `translateX(${offset}px)`;
+}
+
+function updateDots() {
+  const dots = document.querySelectorAll('.dot');
+  dots.forEach((dot, index) => {
+    dot.classList.toggle('active', index === slideIndex);
+  });
+}
+
+window.addEventListener('resize', updateSlidePosition);
+
+// indicators //
+
+function updateDots() {
+  const dots = document.querySelectorAll('.dot');
+  dots.forEach((dot, index) => {
+    dot.classList.toggle('active', index === slideIndex);
+  });
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  document.querySelector('.prev').addEventListener('click', () => moveSlide(-1));
+  document.querySelector('.next').addEventListener('click', () => moveSlide(1));
```

Comment:
Similar a un comentario anterior, usa clases `js-`
```suggestion
  document.querySelector('.js-prev').addEventListener('click', () => moveSlide(-1));
  document.querySelector('.js-next').addEventListener('click', () => moveSlide(1));
```